xonsh/history.py

```diff
@@ -115,14 +115,14 @@ def unlocked_files(self):
         for f in fs:
             try:
                 lj = lazyjson.LazyJSON(f, reopen=False)
-                if lj['locked']:
-                    continue
+                if only_unlocked:
+                    if lj['locked']:
+                        continue
                 # info: closing timestamp, number of commands, filename
                 files.append((lj['ts'][1], len(lj.sizes['cmds']) - 1, f))
                 lj.close()
             except (IOError, OSError, ValueError):
                 continue
-        files.sort()
```

Comment:
> For garbage collect the order of the files does not matter, does it?

Given how the gc is currently implemented, it looks like the sorting does matter.  I think it would be much more complex to reimplement the same behaviour without sorting.  To make this work, I would do something like:

```python
files.append((lj['ts'][1] or time.time(), len(lj.sizes['cmds']) - 1, f))
```

to set the sort time to the current time.